bahmnicore-api/src/main/java/org/bahmni/module/bahmnicore/contract/patient/search/PersonAttributeQueryHelper.java

```diff
@@ -50,10 +50,10 @@ public String appendToJoinClause(String join) {
             join += " LEFT OUTER JOIN person_attribute pattr_results on pattr_results.person_id = p.person_id and pattr_results.voided = 0"
                     + " and  pattr_results.person_attribute_type_id in (" + searchAttributeIds + ") "
                     + " LEFT OUTER JOIN person_attribute_type attrt_results on attrt_results.person_attribute_type_id = pattr_results.person_attribute_type_id "
-                    + " LEFT OUTER JOIN concept_name cn on convert(cn.concept_id,char) = pattr_results.value and cn.concept_name_type = 'FULLY_SPECIFIED'"
+                    + " LEFT OUTER JOIN concept_name cn on cn.concept_id = pattr_results.value and cn.concept_name_type = 'FULLY_SPECIFIED'"
```

Comment:
I am not sure why this would cause slowness in executing query! the reason for using the convert was that the "person_attribute.value" is actually a char field storing the integer of concept id. without a convert, I am guessing mysql anyway would have to a convertion .. but maybe mysql internal convertions are better than using the function! I am approving this PR, but ideally this should be further analyzed.

Reply:
@angshu even i don't find any google threads/info around convert() method having performance impact on the query execution time. But when we execute the query directly on the mysql client we see the clear difference in the query execution time with(37sec) and without(0.01ms) convert method.
I will add these observation to the card with sql query as a reference.

Reply:
tried with cast() which is equivalent to convert(). The query execution time is same for both the methods

Reply:
@angshu @binduak 
Query Execution Plan for original query - 
..... left join openmrs.concept_name cn on **(((cast(openmrs.cn.concept_id as char charset utf8mb4) = convert(openmrs.pattr_results.value using utf8mb4))** and (openmrs.cn.concept_name_type = 'FULLY_SPECIFIED') and (openmrs.attrt_results.format = 'org.openmrs.Concept') and (openmrs.cn.locale = 'en'))) left join openmrs.concept_name def_loc_cn on **(((cast(openmrs.def_loc_cn.concept_id as char charset utf8mb4) = convert(openmrs.pattr_results.value using utf8mb4))** and (openmrs.def_loc_cn.concept_name_type = 'FULLY_SPECIFIED') and (openmrs.attrt_results.format = 'org.openmrs.Concept') and (openmrs.def_loc_cn.locale = 'en'))) where .....


Query Execution Plan for Updated query - 
..... left join openmrs.concept_name cn on **(((openmrs.cn.concept_id = openmrs.pattr_results.value)** and (openmrs.cn.concept_name_type = 'FULLY_SPECIFIED') and (openmrs.attrt_results.format = 'org.openmrs.Concept') and (openmrs.cn.locale = 'en'))) left join openmrs.concept_name def_loc_cn on **(((openmrs.def_loc_cn.concept_id = openmrs.pattr_results.value)** and (openmrs.def_loc_cn.concept_name_type = 'FULLY_SPECIFIED') and (openmrs.attrt_results.format = 'org.openmrs.Concept') and (openmrs.def_loc_cn.locale = 'en'))) where .....

Complete Execution Plan for original query - https://pastebin.com/FF6PRMkZ

Execution of CAST(...), CONVERT(...) functions multiple times are making difference in execution time.